1.16.1/SNP_DirectIP/src/DirectIP/SettingsEnvironment.cpp

```diff
@@ -0,0 +1,41 @@
+#include "SettingsEnvironment.h"
+
+void SettingsEnvironment::FetchEnvironmentVariable(LPCTSTR name, std::string& res) {
+	DWORD bufferSize = 0xFFFF; //Limit according to http://msdn.microsoft.com/en-us/library/ms683188.aspx
+	res.resize(bufferSize);
+	bufferSize = GetEnvironmentVariableA(name, &res[0], bufferSize);
+	if (!bufferSize) {
+		ok = false;
+		res = "";
+		return;
+	}
+	res.resize(bufferSize);
```

Comment:
Would this work? (I don't mind the smaller changes, just the logic of the allocation)

```cpp
DWORD bufferSize = GetEnvironmentVariable(name, nullptr, 0) + 1;
if (bufferSize > 0) {
    res.resize(bufferSize);
    bufferSize = GetEnvironmentVariable(name, res.data(), res.size()) + 1;
    res.pop_back(); // Get rid of null terminator
}

if (bufferSize != 0) {
  ok = false;
  res = "";
}
```

[doc](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getenvironmentvariable) states output is optional and returns the number of characters needed not including the terminating null character. Also shouldn't need the `A` I think?

Reply:
Yes, it seems like it. However this:
```
if (bufferSize != 0) {
  ok = false;
  res = "";
}
```
has its logic inverted. It will be just:
```
...
else {
  ok = false;
  res = "";
}
```

Will fix

Reply:
Sorry yeah that was meant to be == 0